### PR TITLE
Move krel push logic into release API

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This allows us to re-use the same logic in other golang projects like
kubetest.
#### Which issue(s) this PR fixes:

Refers to #1578 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/release/push.go [hunk 1]
```diff
@@ -0,0 +1,324 @@
+package release
```
Comment (justaugustus) [inline pkg/release/push.go:1]: Needs boilerplate

**Review by justaugustus — CHANGES_REQUESTED** [review]
One nit; LGTM otherwise

File: pkg/release/push.go (hunk 1 above)
Comment (saschagrunert) [reply pkg/release/push.go:17]: Fixed, thanks for the hint! :pray:

File: pkg/release/push.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (justaugustus) [inline pkg/release/push.go:2]:
```suggestion
Copyright 2020 The Kubernetes Authors.
```

Comment (saschagrunert) [reply pkg/release/push.go]: Ah yeah, fixed too :+1:

**Comment by justaugustus** [comment]
/lgtm
/approve

Thanks for the quick turnaround on integrating this!

**Pull request merged**